src/middlewared/middlewared/job.py

```diff
@@ -183,7 +183,8 @@ def __init__(self, maxlen=1000):
         self.maxlen = maxlen
         self.count = 0
         self.__dict = OrderedDict()
-        shutil.rmtree(LOGS_DIR)
+        if os.path.exists(LOGS_DIR):
```

Comment:
`with contextlib.suppress(FileNotFoundError)` otherwise it's a potential race condition